### PR TITLE
bcr_reviewer: wait until all runs of the dismiss_approvals workflow are completed before reviewing PRs

### PR DESCRIPTION
**Pull request by meteorcloudy**
Addressing https://github.com/bazelbuild/bazel-central-registry/pull/1621#issuecomment-1994504063

### CODE REVIEW

**Comment by meteorcloudy** [comment]
/cc @fmeum @mihaimaruseac

This was tested at https://github.com/meteorcloudy/bazel-central-registry/actions/runs/8268142699/job/22620838494 where I made the dismiss_approval workflow sleep for 60s.

**Comment by mihaimaruseac** [comment]
This seems reasonable to me. Thank you!

File: actions/bcr-pr-reviewer/index.js [hunk 1]
```diff
@@ -323,9 +323,61 @@ async function runNotifier(octokit) {
...
+  const dismissApprovalsWorkflow = workflows.data.workflows.find(workflow => workflow.path === '.github/workflows/dismiss_approvals.yml');
+  if (!dismissApprovalsWorkflow) {
+    setFailed('The dismiss_approvals workflow is not found');
+    return false;
+  }
+  console.log(`Found dismiss_approvals workflow: ${dismissApprovalsWorkflow.id}`);
+
+  // Wait until all runs of the dismiss_approvals workflow are completed
+  // https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#list-workflow-runs-for-a-repository
+  shouldWaitStatues = ['queued', 'in_progress', 'requested', 'pending', 'waiting'];
```
[the remark this answers is not in the document]
Comment (meteorcloudy) [reply actions/bcr-pr-reviewer/index.js:343]: According to [this doc](https://docs.github.com/en/rest/guides/using-the-rest-api-to-interact-with-checks?apiVersion=2022-11-28#about-check-suites), `cancelled` and `timed_out` are actually conclusions when status is `completed`.  Also there is no easy way to query all workflow runs by a specific PR number. I'll have to get all historical runs to get that information.

However, I think we can add `dismiss_approvals` as a presubmit check, so it actually failed for any reason, the presubmit won't get merged anyway. Does that sound good?

[the remark this answers is not in the document]
Comment (fmeum) [reply actions/bcr-pr-reviewer/index.js:343]: Nice, that's much simpler!

[the remark this answers is not in the document]
Comment (meteorcloudy) [reply actions/bcr-pr-reviewer/index.js:343]: Great!

**Pull request merged**